tests/tests_fabric/parity/test_parity_ddp.py

```diff
@@ -0,0 +1,164 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import time
+from copy import deepcopy
+
+import pytest
+import torch
+import torch.distributed
+import torch.nn.functional
+from torch.nn.parallel.distributed import DistributedDataParallel
+from torch.utils.data import DataLoader
+from torch.utils.data.distributed import DistributedSampler
+
+from lightning.fabric.fabric import Fabric
+from tests_fabric.helpers.runif import RunIf
+from tests_fabric.parity.models import ConvNet
+from tests_fabric.parity.utils import (
+    cuda_reset,
+    is_cuda_memory_close,
+    is_state_dict_equal,
+    is_timing_close,
+    make_deterministic,
+)
+
+
+def train_torch_ddp(
+    rank,
+    world_size,
+    device=torch.device("cpu"),
+):
+    make_deterministic()
+    memory_stats = {}
+
+    os.environ["LOCAL_RANK"] = str(rank)
+    if torch.distributed.is_available() and not torch.distributed.is_initialized():
+        torch.distributed.init_process_group("gloo", rank=rank, world_size=world_size)
```

Comment:
The device is customizable (in the signature) but the backend is not. This could become an issue.

Either make it configurable or make it choose based on the device

Reply:
gloo works for both cpu and gpu, which is why I chose this. It is fine to keep it like this for now

Reply:
Doeesn't Fabric choose NCCL? Wouldn't that be a source of differences when comparing them?

Reply:
Sure, then let's close the PG and recreate it but keep the same processes from Fabric.

Reply:
it wasn't the source of the difference because as you pointed out, the if-statement wouldn't be true